dev-docs/RFCs/v8.0/ui-module-rfc.md

```diff
@@ -0,0 +1,137 @@
+# UI Module RFC
+
+* **Author**: Xiaoji Chen
+* **Date**: December 2022
+* **Status**: **Pending Review**
+
+## Overview
+
+Visualization applications often require some chrome around the WebGL canvas to offer better user experience. While it is possible for every app to implement their own UI components that interact with deck.gl, it is no trivial effort to implement them well. 
+
+This RFC proposes to add a new module that packages common UI components that seamlessly integrate with deck.gl, including but not limited to:
+
+- Marker: HTML element pinned to a location in the world space
+- Navigation: buttons to zoom / rotate / reset the viewport
+- Fullscreen: button to enter / exit fullscreen mode
+- Geolocate: button to track the user's location using the browser's Geolocation API
+- Legend: visual comparison of sizes, colors etc. corresponding to the rendered layers and viewport
+
+
+## Background
+
+Deck.gl does not currently export any DOM-based UI components. The documentation recommends react-map-gl as a companion library in the following pattern:
+
+```jsx
+import {StaticMap, MapContext, NavigationControl, FullscreenControl} from 'react-map-gl'
+
+<DeckGL
+  ContextProvider={MapContext.Provider}
+>
+  <StaticMap />
+  <NavigationControl />
```

Comment:
How would this work with multiple views? Will it be possible to attach multiple `NavigationControl`s

Reply:
With React: https://deck.gl/docs/api-reference/react/deckgl#jsx-views

Without React: not sure, maybe we can support an additional parameter when adding the control:

```js
new Deck({
  uiControls: [
    new NavigationControl({viewId: 'map'})
  ]
})

// or
deck.addUIControl(control: UIControl, viewId?: string)
```

Reply:
Nice